src/System.Runtime.InteropServices/src/System/Runtime/InteropServices/RuntimeEnvironment.cs

```diff
@@ -23,7 +23,12 @@ public static bool FromGlobalAccessCache(System.Reflection.Assembly a)
         }
         public static string GetRuntimeDirectory()
         {
-            return Path.GetDirectoryName(typeof(object).Assembly.Location) + Path.DirectorySeparatorChar;
+            string runtimeDirectory = typeof(object).Assembly.Location;
+            if(string.IsNullOrEmpty(runtimeDirectory) || !Path.IsPathRooted(runtimeDirectory))
+            { 
+                return Path.GetDirectoryName(AppDomain.CurrentDomain.BaseDirectory) + Path.DirectorySeparatorChar;
```

Comment:
You can just do `runtimeDirectory = AppDomain.CurrentDomain.BaseDirectory` here to avoid duplicated code.